src/BaseElement.js

```diff
@@ -100,9 +100,7 @@ class BaseElement extends HTMLElement {
 		// remove events for elements in the component
 		this.removeEvents();
 
-		if (this._options.provideContext) {
-			this.removeEventListener('request-context', this.onRequestContext);
-		}
+		this.removeEventListener('request-context', this.onRequestContext);
```

Comment:
@quarkus do we actually need to remove this one? Since the element will be gone anyways. I think all the event listeners will be removed along with it.

Reply:
yeah .. thats probably true  but we are cleaning up the other ones as well. it was mor a just in case thing tbh.